torch/_decomp/decompositions.py

```diff
@@ -1710,8 +1710,8 @@ def native_layer_norm_backward(
 
     return (
         _maybe_cast(d_input, input.dtype),
-        _maybe_cast(d_weight, input.dtype),
-        _maybe_cast(d_bias, input.dtype),
+        _maybe_cast(d_weight, weight.dtype if weight is not None else None),
```

Comment:
@albanD is this right ?

Reply:
Yes it is correct.
It's because the weight and bias are optional. But when they're not there, both these gradients will be None anyways so no cast is needed.